BorrowBookControl.java

```diff
@@ -2,116 +2,116 @@
-		}	
-		BOOK = LIBRARY.Book(bookId);
-		if (BOOK == null) {
-			UI.Display("Invalid bookId");
-			return;
-		}
-		if (!BOOK.AVAILABLE()) {
-			UI.Display("Book cannot be borrowed");
-			return;
-		}
-		PENDING.add(BOOK);
-		for (book B : PENDING) {
-			UI.Display(B.toString());
-		}
-		if (LIBRARY.Loans_Remaining_For_Member(M) - PENDING.size() == 0) {
-			UI.Display("Loan limit reached");
-			Complete();
-		}
-	}
-	
-	
-	public void Complete() {
-		if (PENDING.size() == 0) {
-			cancel();
-		}
-		else {
-			UI.Display("\nFinal Borrowing List");
-			for (book B : PENDING) {
-				UI.Display(B.toString());
-			}
-			COMPLETED = new ArrayList<loan>();
-			UI.Set_State(BorrowBookUI.UI_STATE.FINALISING);
-			State = CONTROL_STATE.FINALISING;
-		}
-	}
-
-
-	public void Commit_LOans() {
-		if (!State.equals(CONTROL_STATE.FINALISING)) {
-			throw new RuntimeException("BorrowBookControl: cannot call commitLoans except in FINALISING state");
-		}	
-		for (book B : PENDING) {
-			loan LOAN = LIBRARY.ISSUE_LAON(B, M);
-			COMPLETED.add(LOAN);			
-		}
-		UI.Display("Completed Loan Slip");
-		for (loan LOAN : COMPLETED) {
-			UI.Display(LOAN.toString());
-		}
-		UI.Set_State(BorrowBookUI.UI_STATE.COMPLETED);
-		State = CONTROL_STATE.COMPLETED;
-	}
-
-	
-	public void cancel() {
-		UI.Set_State(BorrowBookUI.UI_STATE.CANCELLED);
-		State = CONTROL_STATE.CANCELLED;
-	}
-	
-	
+
+    private BorrowBookUI UI;
+
+    private library library;
+    private member member;
+    private enum ControlState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+    private ControlState state;
+
+    private List<book> pending;
+    private List<loan> completed;
+    private book book;
+
+
+    public BorrowBookControl() {
+        this.library = library.INSTANCE();
+        this.state = ControlState.INITIALISED;
+    }
+
+
+    public void setUI(BorrowBookUI UI) {
+        if (!this.state.equals(ControlState.INITIALISED))
+            throw new RuntimeException("BorrowBookControl: cannot call setUI except in INITIALISED state");
+
+        this.UI = UI;
+        UI.setState(BorrowBookUI.UIState.READY);
+        this.state = ControlState.READY;
+    }
+
+
+    public void swiped(int memberId) {
+        if (!this.state.equals(ControlState.READY))
+            throw new RuntimeException("BorrowBookControl: cannot call cardSwiped except in READY state");
+
+        this.member = library.MEMBER(memberId);
+        if (this.member == null) {
+            this.UI.display("Invalid memberId");
+            return;
+        }
+        if (this.library.MEMBER_CAN_BORROW(this.member)) {
+            this.pending = new ArrayList<>();
+            this.UI.setState(BorrowBookUI.UIState.SCANNING);
+            this.state = ControlState.SCANNING; }
+        else
+        {
+            this.UI.display("Member cannot borrow at this time");
+            this.UI.setState(BorrowBookUI.UIState.RESTRICTED); }}

```

Comment:
D13: refer to correct bracketing style. final '}' should be on next line.
Please look over other lines in both classes to ensure they also meet this standard.

Reply:
interesting, the opening bracket shows correctly placed but the closing incorrectly in IDE. Fixed closing bracket